scipy/linalg/tests/test_extending.py

```diff
@@ -0,0 +1,81 @@
+import os
+import platform
+import shutil
+import subprocess
+import sys
+import sysconfig
+from importlib.util import module_from_spec, spec_from_file_location
+
+import numpy as np
+import pytest
+
+from scipy.linalg.blas import cdotu # type: ignore[attr-defined]
+from scipy.linalg.lapack import dgtsv # type: ignore[attr-defined]
+
+try:
+    # Need type: ignore[import-untyped] for mypy >= 1.6
+    import cython # type: ignore[import]
+    from Cython.Compiler.Version import ( # type: ignore[import]
+        version as cython_version)
+except ImportError:
+    cython = None
+else:
+    from scipy._lib import _pep440
+
+    # Note: keep in sync with the one in pyproject.toml
+    required_version = '3.0.8'
+    if _pep440.parse(cython_version) < _pep440.Version(required_version):
+        # too old or wrong cython, skip the test
+        cython = None
+
+
+@pytest.mark.skipif(platform.machine() in ["wasm32", "wasm64"],
+                    reason="Can't start subprocess")
+@pytest.mark.skipif(cython is None, reason="requires cython")
+@pytest.mark.slow
+def test_cython(tmp_path):
+    # build the examples in a temporary directory
+    srcdir = os.path.join(os.path.dirname(__file__), '..')
+    shutil.copytree(srcdir, tmp_path / 'linalg')
+    build_dir = tmp_path / 'linalg' / 'tests' / '_cython_examples'
+    target_dir = build_dir / 'build'
+    os.makedirs(target_dir, exist_ok=True)
+    if sys.platform == "win32":
+        subprocess.check_call(["meson", "setup",
+                               "--buildtype=release", 
+                               "--vsenv", str(build_dir)],
+                              cwd=target_dir,
+                              )
+    else:
+        subprocess.check_call(["meson", "setup", str(build_dir)],
+                              cwd=target_dir
+                              )
+    subprocess.check_call(["meson", "compile", "-vv"], cwd=target_dir)
+
+    # import without adding the directory to sys.path
+    suffix = sysconfig.get_config_var('EXT_SUFFIX')
+
+    def load(modname):
+        so = (target_dir / modname).with_suffix(suffix)
+        spec = spec_from_file_location(modname, so)
+        mod = module_from_spec(spec)
+        spec.loader.exec_module(mod)
+        return mod
+
+    # test that the module can be imported
+    extensions = load("extending")
```

Comment:
This logic could presumably be put in a function in `_lib`?

Reply:
Great idea! I refactored it into a function in `scipy._lib._testutils`. This resolves most of my concerns regarding code duplication.